File: services/aas-designer/AasDesignerApi/Model/Client/SubmodelResult.cs
using BaSyx.Models.Core.AssetAdministrationShell.Implementations;

namespace AasDesignerApi.Model.Client
{
    public class SubmodelResult
    {
        public List<Submodel> Submodels { get; set; } = new List<Submodel>();
        public List<BaSyx.Models.Core.AssetAdministrationShell.Semantics.ConceptDescription> ConceptDescriptions { get; set; } =
        [];

        public List<string> V3SubmodelsPlain { get; set; } = new List<string>();
        public List<string> V3ConceptDescriptionsPlain { get; set; } = new List<string>();
    }
}

File: frontend/libs/aas-designer-core/src/lib/generator/model/submodel-result.ts
import * as aas from '@aas-core-works/aas-core3.1-typescript';
import { ConceptDescription, Submodel } from '@aas/model';

export class SubmodelResult {
  submodels: Submodel[] = [];
  conceptDescriptions: ConceptDescription[] = [];

  v3Submodels: aas.types.Submodel[] = [];
  v3ConceptDescriptions: aas.types.ConceptDescription[] = [];

  v3SubmodelsPlain: string[] = [];
  v3ConceptDescriptionsPlain: string[] = [];

  static fromDto(dto: SubmodelResult) {
    const res = new SubmodelResult();

    for (const sm of dto.submodels ?? []) {
      res.submodels.push(Submodel.fromDto(sm));
    }
    for (const cd of dto.conceptDescriptions ?? []) {
      res.conceptDescriptions.push(ConceptDescription.fromDto(cd));
    }

    for (const sm of dto.v3SubmodelsPlain ?? []) {
      const instanceOrErrorPlain = aas.jsonization.submodelFromJsonable(JSON.parse(sm));
      if (instanceOrErrorPlain.value != null) {
        res.v3Submodels.push(instanceOrErrorPlain.value);
      }
    }
    for (const cd of dto.v3ConceptDescriptionsPlain ?? []) {
      const instanceOrErrorPlain = aas.jsonization.conceptDescriptionFromJsonable(JSON.parse(cd));
      if (instanceOrErrorPlain.value != null) {
        res.v3ConceptDescriptions.push(instanceOrErrorPlain.value);
      }
    }

    return res;
  }
}
